source/core/grids.cc

```diff
@@ -109,6 +110,142 @@ attach_grid_to_triangulation(
             mesh_parameters.grid_arguments);
         }
     }
+  // Customizable cylinder mesh
+  else if (mesh_parameters.type == Parameters::Mesh::Type::cylinder)
+    {
+      if (mesh_parameters.simplex)
+        {
+          throw std::runtime_error(
+            "Unsupported mesh type - custom cylinder mesh with simplex is not supported, use dealii cylinder prior using simplex mesh.");
+        }
+      else if (dim != 3)
+        {
+          throw std::runtime_error(
+            "Unsupported mesh type - custom cylinder mesh is only supported in 3d.");
+        }
+
+      if constexpr (dim == 3)
+        {
+          // Separate arguments of the string
+          std::vector<std::string> arguments;
+          std::stringstream        s_stream(mesh_parameters.grid_arguments);
+          while (s_stream.good())
+            {
+              std::string substr;
+              getline(s_stream, substr, ':');
+              arguments.push_back(substr);
+            }
+
+          // Arguments declaration
+          unsigned int subdivisions;
+          double       radius, half_height;
+          if (arguments.size() != 3)
+            {
+              throw std::runtime_error(
+                "Mandatory customizable cylinder parameters are (x subdivisions: radius : half height)");
+            }
+          else
+            {
+              std::vector<double> arguments_double =
+                dealii::Utilities::string_to_double(arguments);
+              subdivisions = static_cast<int>(arguments_double[0]);
+              radius       = arguments_double[1];
+              half_height  = arguments_double[2];
+            }
+
+          if (mesh_parameters.grid_type == "classic")
+            {
+              // Create a subdivided cylinder from deal.ii
+              GridGenerator::subdivided_cylinder(*triangulation,
+                                                 subdivisions,
+                                                 radius,
+                                                 half_height);
+            }
+          else
+            {
+              // Create a temporary 2d mesh
+              Triangulation<dim - 1, spacedim - 1> temporary_triangulation;
```

Comment:
I would put 2 here instead of dim-1, because it is explicitly a 2d mesh. Not a big deal, but I find defensive programming always important.

Reply:
Yeah, I was ambiguous about it, I will change all `dim` and `dim - 1` to 3 and 2 in that code